src/Products/Viewer/Controllers/ViewerApiController.cs

```diff
@@ -430,149 +453,125 @@ private LoadDocumentEntity LoadDocument(PostedDataEntity postedData, bool loadAl
 
-        private void SetOptions(HtmlOptions options, string password, int pageNumber)
+        private List<string> GetAllPagesContent(string password, string documentGuid, IList<Page> pages)
         {
-            Watermark watermark = null;
-            if (!String.IsNullOrEmpty(globalConfiguration.Viewer.GetWatermarkText()))
-            {
-                // Set watermark properties
-                watermark = new Watermark(globalConfiguration.Viewer.GetWatermarkText());
-                watermark.Color = System.Drawing.Color.Blue;
-                watermark.Position = WatermarkPosition.Diagonal;
-                watermark.Width = 100;
-            }
-            options.EmbedResources = true;
-            // set password for protected document
-            if (!string.IsNullOrEmpty(password))
-            {
-                options.Password = password;
-            }
-            if (watermark != null)
-            {
-                options.Watermark = watermark;
-            }
-            if (pageNumber != 0)
-            {
-                options.PageNumber = pageNumber;
-                options.CountPagesToRender = 1;
+            List<string> allPages = new List<string>();
+
+            for (int i = 0; i < pages.Count; i++)
+            {               
+                allPages.Add(RenderPageToString(pages[i].Number, password, documentGuid));
             }
-            options.CellsOptions.ShowGridLines = globalConfiguration.Viewer.GetShowGridLines();
+
+            return allPages;
         }
 
-        private void SetOptions(ImageOptions options, string password, int pageNumber)
+        private void SetWatermarkOptions(ViewOptions options)
         {
             Watermark watermark = null;
-            if (!String.IsNullOrEmpty(globalConfiguration.Viewer.GetWatermarkText()))
+            if (!string.IsNullOrEmpty(globalConfiguration.Viewer.GetWatermarkText()))
             {
                 // Set watermark properties
                 watermark = new Watermark(globalConfiguration.Viewer.GetWatermarkText());
                 watermark.Color = System.Drawing.Color.Blue;
-                watermark.Position = WatermarkPosition.Diagonal;
-                watermark.Width = 100;
-            }
-            // set password for protected document
-            if (!string.IsNullOrEmpty(password))
-            {
-                options.Password = password;
+                watermark.Position = Position.Diagonal;
             }
+
             if (watermark != null)
             {
                 options.Watermark = watermark;
             }
-            if (pageNumber != 0)
-            {
-                options.PageNumber = pageNumber;
-                options.CountPagesToRender = 1;
-            }
-            options.CellsOptions.ShowGridLines = globalConfiguration.Viewer.GetShowGridLines();
+        }
+
+        private LoadOptions GetLoadOptions(string password)
+        {
+            var loadOptions = new LoadOptions() 
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Remove these redundant parentheses.](https://app.codacy.com/manual/GroupDocs/GroupDocs.Viewer-for-.NET-MVC/pullRequest?prid=5242613)